### PR TITLE
Feature/discovery/page details

### PR DESCRIPTION
**Pull request by camillebrulotte**
Hésitez pas si vous avez des idées de bulles à ajouter ou à modifier :D 

https://user-images.githubusercontent.com/37625944/123719998-f7203b00-d850-11eb-87b8-74d5187b70ed.mp4

J'ai aussi testé sur IOS, ça fonctionne :)

### CODE REVIEW

**Comment by apomalyn** [comment]
Just a quick thought. When I tap the schedule tab during the "tour" I should be moved to the schedule page and see the tour of this page, right?

**Comment by camillebrulotte** [comment]
> Just a quick thought. When I tap the schedule tab during the "tour" I should be moved to the schedule page and see the tour of this page, right?

I tried to do it but I was not able to move to other pages ;( The library seems to work more like a "page to page" discovery than a "walkthrough"... Unless you have an idea on how to do it? :D

File: lib/l10n/intl_fr.arb [hunk 1]
```diff
@@ -188,16 +190,33 @@
...
+  "discovery_page_student_grades_grade_button": "Ici vous voyez votre note actuelle. En tapant dessus, vous pourrez voir le détail de toutes les notes obtenues dans le cours.",
+  
+  "discovery_page_student_profile": "Vous trouvez ici toutes les informations reliées à votre compte étudiant, de la balance à vos programmes actifs en passant par votre code permanent.",
+  
+  "discovery_navbar_ets_title": "ÉTS",
+  "discovery_navbar_ets_details": "La page ÉTS vous permet de consulter rapidement tous les liens importants de l’ÉTS, tels que Moodle,  MonÉTS et les mesures de sécurité.",
+  
+  "discovery_navbar_more_title": "Plus",
+  "discovery_navbar_more_details": "La page Plus vous permet d'avoir accès aux options supplémentaires de l'application.",
+  "discovery_page_more_report_bug": "L'application n'est pas parfaite et nous sommes toujours ouverts à des suggestions! Si vous voyez un problème ou si vous souhaitez nous faire partie d'une idée, vous pouvez utiliser cette fonctionnalité",
```
Comment (MysticFragilist) [inline lib/l10n/intl_fr.arb:215]:
```suggestion
  "discovery_page_more_report_bug": "L'application n'est pas parfaite et nous sommes toujours ouverts à des suggestions! Si vous voyez un problème ou si vous souhaitez nous faire part d'une idée, vous pouvez utiliser cette fonctionnalité",
```

File: lib/l10n/intl_fr.arb [hunk 2]
```diff
@@ -188,16 +190,33 @@
...
+  "discovery_navbar_student_title": "Étudiant",
+  "discovery_navbar_student_details": "La page Étudiant affiche vos notes et les informations de votre profil.", 
+  
+  "discovery_page_student_grades_session": "Vos cours sont divisés par session, avec la plus récente en haut, en tout temps vous pouvez tirer sur la liste pour la rafraichir.",
+  "discovery_page_student_grades_grade_button": "Ici vous voyez votre note actuelle. En tapant dessus, vous pourrez voir le détail de toutes les notes obtenues dans le cours.",
+  
+  "discovery_page_student_profile": "Vous trouvez ici toutes les informations reliées à votre compte étudiant, de la balance à vos programmes actifs en passant par votre code permanent.",
+  
+  "discovery_navbar_ets_title": "ÉTS",
+  "discovery_navbar_ets_details": "La page ÉTS vous permet de consulter rapidement tous les liens importants de l’ÉTS, tels que Moodle,  MonÉTS et les mesures de sécurité.",
```
Comment (MysticFragilist) [inline lib/l10n/intl_fr.arb:211]:
```suggestion
  "discovery_navbar_ets_details": "La page ÉTS vous permet de consulter rapidement tous les liens importants de l’ÉTS, tels que Moodle, MonÉTS et les mesures de sécurité.",
```

File: lib/l10n/intl_en.arb [hunk 3]
```diff
@@ -190,16 +192,33 @@
...
+  "discovery_page_student_grades_grade_button": "Here you see your current grades. By tapping on it, you will be able to see the details of all the evaluations obtained in the course.",
+  
+  "discovery_page_student_profile": "Here you will find all the information related to your student account, from the balance to your active programs through your permanent code.",
+  
+  "discovery_navbar_ets_title": "ÉTS",
+  "discovery_navbar_ets_details": "The ÉTS page allows you to quickly consult all of the important ÉTS links, such as Moodle, MonÉTS and security measures.",  
+  
+  "discovery_navbar_more_title": "More",
+  "discovery_navbar_more_details": "The More page gives you access to additional options of the application.",
+  "discovery_page_more_report_bug": "The app is not perfect and, we are always open to suggestions! If you see a problem or want us to be part of an idea, you can use this feature.",
```
Comment (MysticFragilist) [inline lib/l10n/intl_en.arb:217]:
```suggestion
  "discovery_page_more_report_bug": "The app is not perfect and we are always open to suggestions! If you see a problem or want us to be part of an idea, you can use this feature.",
```

File: lib/l10n/intl_en.arb [hunk 4]
```diff
@@ -190,16 +192,33 @@
...
+  
+  "discovery_navbar_schedule_title": "Schedule",
+  "discovery_navbar_schedule_details": "The Schedule page allows you to view the events of your school calendar.",
+  "discovery_page_schedule_settings": "In the settings, you can change the calendar format and the first day of the week displayed.",
+  
+  "discovery_navbar_student_title": "Student",
+  "discovery_navbar_student_details": "The Student page displays your grades and profile information.",
+  
+  "discovery_page_student_grades_session": "Your courses are divided by session, with the most recent at the top. At any moment, you can drag the list to refresh it.",
+  "discovery_page_student_grades_grade_button": "Here you see your current grades. By tapping on it, you will be able to see the details of all the evaluations obtained in the course.",
```
Comment (MysticFragilist) [inline lib/l10n/intl_en.arb:208]: More of an open question, but should we do a discovery on the page grade details? @camillebrulotte @apomalyn @AlexandreMarkus

File: lib/l10n/intl_en.arb [hunk 5]
```diff
@@ -190,16 +192,33 @@
...
+  
+  "discovery_page_student_profile": "Here you will find all the information related to your student account, from the balance to your active programs through your permanent code.",
+  
+  "discovery_navbar_ets_title": "ÉTS",
+  "discovery_navbar_ets_details": "The ÉTS page allows you to quickly consult all of the important ÉTS links, such as Moodle, MonÉTS and security measures.",  
+  
+  "discovery_navbar_more_title": "More",
+  "discovery_navbar_more_details": "The More page gives you access to additional options of the application.",
+  "discovery_page_more_report_bug": "The app is not perfect and, we are always open to suggestions! If you see a problem or want us to be part of an idea, you can use this feature.",
+  "discovery_page_more_contributors": "Here you will find the list of all the students who worked on the project from a code point of view.",
```
Comment (MysticFragilist) [inline lib/l10n/intl_en.arb:218]: I find that the lasts few words worded a bit weird, maybe if we say "Here you will find the list of all the students who contributed and developed this application." would it be better? any other idea?

File: lib/core/viewmodels/schedule_viewmodel.dart [hunk 6]
```diff
@@ -162,4 +167,19 @@ class ScheduleViewModel extends FutureViewModel<List<CourseActivity>> {
...
+
+  Future<void> startDiscovery(BuildContext context) async {
+    if (await _settingsManager.getString(PreferencesFlag.discoverySchedule) ==
+        null) {
+      final List<String> ids =
+          findDiscoveriesByGroupName(context, DiscoveryGroupIds.pageSchedule)
+              .map((e) => e.featureId)
+              .toList();
+
+      Future.delayed(const Duration(milliseconds: 500),
```
Comment (MysticFragilist) [inline lib/core/viewmodels/schedule_viewmodel.dart:179]: Should we make this 1 seconds like on More page? I think it should be nice for the user to see the page fully loaded a bit before launching the discovery? @AlexandreMarkus @apomalyn @camillebrulotte your thoughts?

File: pubspec.yaml [hunk 7]
```diff
@@ -15,7 +15,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 # In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
 # Read more about iOS versioning at
 # https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
-version: 4.1.0+1
+version: 4.2.0+1
```
Comment (MysticFragilist) [inline pubspec.yaml:18]: This isn't related to this PR, but @apomalyn  what is the +1 after the version? I keep wondering what it does

**Review by MysticFragilist — CHANGES_REQUESTED** [review]
Niice job!!! 😄

File: lib/l10n/intl_en.arb (hunk 5 above)
Comment (camillebrulotte) [reply lib/l10n/intl_en.arb:218]: Bien vue, je devais être fatiguée quand j'ai écris ça xD

File: lib/core/viewmodels/schedule_viewmodel.dart (hunk 6 above)
Comment (camillebrulotte) [reply lib/core/viewmodels/schedule_viewmodel.dart:179]: Sure ! Mais dans more page c'est la même affaire :D

File: pubspec.yaml (hunk 7 above)
Comment (apomalyn) [reply pubspec.yaml:18]: aha let me bring light to these numbers :smile: this is the build number or build metadata ^^
If you want to read the complete version of SemVer head [here](https://semver.org/) :smile: if you need me to complete my answer don't hesitate to ask :smile:

File: lib/l10n/intl_en.arb (hunk 4 above)
Comment (apomalyn) [reply lib/l10n/intl_en.arb:208]: I do agree with you :smile:

**Comment by apomalyn** [comment]
> > Just a quick thought. When I tap the schedule tab during the "tour" I should be moved to the schedule page and see the tour of this page, right?
> 
> I tried to do it but I was not able to move to other pages ;( The library seems to work more like a "page to page" discovery than a "walkthrough"... Unless you have an idea on how to do it? :D

:thinking: I will need to try and do some research but for now I'm fully concentrated on the release :sweat_smile:

File: pubspec.yaml (hunk 7 above)
Comment (MysticFragilist) [reply pubspec.yaml:18]: Ohh nice! Thanks for the info 😄

File: lib/l10n/intl_en.arb (hunk 4 above)
Comment (camillebrulotte) [reply lib/l10n/intl_en.arb:208]: @apomalyn @MysticFragilist Enfait vous pensiez à mettre quoi comme description dans la grade details ? :) 

Autre que les cercles de progress bar qui pourraient être ambigu et les evaluation tile, je sais pas trop quoi expliquer d'autre.

**Comment by camillebrulotte** [comment]
J'ai terminé les corrections et j'ai ajouté le discovery sur la page de détails. 

Toutefois, quand on clique sur le CircularProgressBar pour fermer le discovery, il y a un RenderFlex qui est probablement causé par la librairie et elle n'a pas l'air d'être bien maintenue. 

Donc, pendant 1 miliseconde on peut voir : 
image

File: lib/ui/widgets/grade_evaluation_tile.dart [hunk 8]
```diff
@@ -78,20 +83,23 @@ class _GradeEvaluationTileState extends State<GradeEvaluationTile>
...
-                      studentGrade: Utils.getGradeInPercentage(
-                        widget.evaluation.mark,
-                        widget.evaluation.correctedEvaluationOutOfFormatted,
-                      ),
-                      averageGrade: Utils.getGradeInPercentage(
-                        widget.evaluation.passMark,
-                        widget.evaluation.correctedEvaluationOutOfFormatted,
-                      ),
-                    );
+                    return _buildDiscoveryFeatureDescriptionWidget(
```
Comment (MysticFragilist) [inline lib/ui/widgets/grade_evaluation_tile.dart:86]: Am I missing something or will this build a discovery item for every evaluation tile? Don't know if it's maybe difficult to put it on one only? (maybe with a parameter you passed to the widget when it is created) 😄

**Review by MysticFragilist — CHANGES_REQUESTED** [review]
Small question, but when address, it's going to be a nice LGTM! 🚀

**Comment by MysticFragilist** [comment]
Oh aussi bonne nouvelle!!! J'ai trouvé dans le code de Feature Discovery comment il traite les sharedPrefs pour chacun des features: 
```dart
@override
Future<void> completeStep(String? featureId) async {
  final prefs = await SharedPreferences.getInstance();
  await prefs.setBool(_normalizeFeatureId(featureId), true);
}

// ...

String _normalizeFeatureId(String? featureId) =>
      '$sharedPrefsPrefix$featureId';
```
(voir [Persistent_provider.dart:56-79](https://github.com/ayalma/feature_discovery/blob/master/lib/src/foundation/persistence_provider.dart))
Ça veut dire qu'on va pouvoir ajouter le bouton de discovery dans les settings! Comme c'est un tout autre sujet qui va nécessiter code/ui/tests, je propose de terminer cette PR et de faire l'ajout du bouton dans une autre.

**Comment by camillebrulotte** [comment]
> Oh aussi bonne nouvelle!!! J'ai trouvé dans le code de Feature Discovery comment il traite les sharedPrefs pour chacun des features:
> 
> ```dart
> @override
> Future<void> completeStep(String? featureId) async {
>   final prefs = await SharedPreferences.getInstance();
>   await prefs.setBool(_normalizeFeatureId(featureId), true);
> }
> 
> // ...
> 
> String _normalizeFeatureId(String? featureId) =>
>       '$sharedPrefsPrefix$featureId';
> ```
> 
> (voir [Persistent_provider.dart:56-79](https://github.com/ayalma/feature_discovery/blob/master/lib/src/foundation/persistence_provider.dart))
> Ça veut dire qu'on va pouvoir ajouter le bouton de discovery dans les settings! Comme c'est un tout autre sujet qui va nécessiter code/ui/tests, je propose de terminer cette PR et de faire l'ajout du bouton dans une autre.

Nice ! Je réouvre la tâche dans le projet Discovery ! Merci :D

**Review by MysticFragilist — DISMISSED** [review]
LGTM! 😄 🍾

**Review by apomalyn — DISMISSED** [review]
LGTM !! 🍾 🍾

File: pubspec.yaml [hunk 9]
```diff
@@ -15,7 +15,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 # In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
 # Read more about iOS versioning at
 # https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
-version: 4.1.4+1
+version: 4.3.0+1
```
Comment (apomalyn) [inline pubspec.yaml:18]: Why `4.3.0` and not `4.2.0`?

File: lib/ui/widgets/grade_evaluation_tile.dart [hunk 10]
```diff
@@ -1,10 +1,15 @@
...
+import 'package:feature_discovery/feature_discovery.dart';
 import 'package:flutter/material.dart';
 import 'package:intl/intl.dart';
 
+// CONSTANTS
+import 'package:notredame/core/constants/discovery_ids.dart';
+
 // MODELS
 import 'package:notredame/core/models/evaluation.dart';
+import 'package:notredame/ui/utils/discovery_components.dart';
```
Comment (JonathanDuvalV) [inline lib/ui/widgets/grade_evaluation_tile.dart:12]: Could you move it to utils?

**Review by MysticFragilist — APPROVED** [review]

**Review by JonathanDuvalV — APPROVED** [review]

**Pull request merged**